modules/apigee/main.tf

```diff
@@ -100,6 +100,13 @@ resource "google_apigee_instance" "instances" {
   consumer_accept_list     = each.value.consumer_accept_list
 }
 
+resource "google_apigee_nat_address" "apigee-nat" {
+  for_each    = local.instances
```

Comment:
I think this doesn't work. You can't use `for_each` and `count` together.

Yo might want to filter `local.instances` with a comprehension.